node/utils/metrics/organization.ts

```diff
@@ -0,0 +1,96 @@
+import type { Logger } from '@vtex/api/lib/service/logger/logger'
+
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+interface UpdateOrganizationFieldsMetric {
+  update_details: { properties: string[] }
+}
+
+type UpdateOrganization = Metric & {
+  fields: UpdateOrganizationFieldsMetric
+}
+
+export interface UpdateOrganizationParams {
+  account: string
+  currentOrganizationData: Organization
+  updatedProperties: UpdatedOrganizationProps
+}
+
+const buildUpdateOrganizationMetric = (
+  account: string,
+  updatedProperties: string[]
+): UpdateOrganization => {
+  const updateOrganizationFields: UpdateOrganizationFieldsMetric = {
+    update_details: { properties: updatedProperties },
+  }
+
+  return {
+    account,
+    description: 'Update Organization Action - Graphql',
+    fields: updateOrganizationFields,
+    kind: 'update-organization-graphql-event',
+    name: 'b2b-suite-buyerorg-data',
+  } as UpdateOrganization
+}
+
+const getFieldsNamesByFieldsUpdated = (
+  updateOrganizationParams: UpdateOrganizationParams
+): string[] => {
+  const updatedPropName: string[] = []
+
+  const { currentOrganizationData, updatedProperties } =
+    updateOrganizationParams
+
+  for (const key in updatedProperties) {
+    if (Object.prototype.hasOwnProperty.call(updatedProperties, key)) {
+      const value = updatedProperties[key as keyof typeof updatedProperties]
+
+      // I tried to compare the objects value !== currentOrganizationData[key as keyof Organization,
+      // but it was not working, so I use JSON.stringify
+      if (
+        JSON.stringify(value) !==
+        JSON.stringify(currentOrganizationData[key as keyof Organization])
+      ) {
+        updatedPropName.push(key)
+      }
+    }
+  }
+
+  return updatedPropName
+}
+
+export interface UpdatedOrganizationProps {
+  priceTables: any[]
```

Comment:
In Typescript we always avoid using `any`. With the right tsconfig this should be throwing lint warnings or even errors actually.

For use cases where the actual detailed type can't be known in advance, or we don't have the definitions anywhere, you can instead use `unknown`, which is stricter than `any` (it doesn't allow accessing fields of the variable, for example, while `any` will allow anything).

In this case though, I believe that you can be even more generic. The actual keys here don't matter much to you, you just need to know that this will be a map of string to unknown values, so this type could just be:

```js
export type UpdatedOrganizationProps = Record<string, unknown>
```

This should also help with the code that was trying to access keys within this object, within `getFieldsNamesByFieldsUpdated`.

Reply:
Or even better, you can just use the `Organization` type directly, shouldn't they be basically the same? The update object is probably the same type, but with all keys being optional. In Typescript you can achieve that like this:

```js
export type UpdateOrganizationProps = Partial<Organization>
```

That will also help when comparing their values, as the keys will now match, and you won't need to cast types around.